disnake/interactions/message.py

```diff
@@ -74,13 +79,33 @@ class MessageInteraction(Interaction):
 
     def __init__(self, *, data: MessageInteractionPayload, state: ConnectionState):
         super().__init__(data=data, state=state)
-        self.data: MessageInteractionData = MessageInteractionData(data=data["data"])
+        self.data: MessageInteractionData = MessageInteractionData(
+            data=data["data"], state=state, guild_id=self.guild_id
+        )
         self.message = Message(state=self._state, channel=self.channel, data=data["message"])
 
+    # TODO: perhaps keep `values` like before and add a `resolved_values` instead?
```

Comment:
unsure about this

Reply:
I think we should make this change be non-breaking and use resolved_values.

Reply:
Hmmm, I'm now unsure too because the new selects have `values` refer to the values chosen and they are the resolved objects.

Reply:
Yeah. For now I've decided to have a separate field, since the typing in existing bot code that uses `MessageInteraction.values` for string selects would otherwise be completely borked.
This way there's also no confusion between `inter.values` and `inter.data.values`.
https://github.com/DisnakeDev/disnake/pull/803/commits/c7ca5ea4db6db971c1cee6afd7d8464983f22d0a